pkg/collector/offline_dump.go

```diff
@@ -0,0 +1,247 @@
+package collector
```

Comment:
I think this needs to go in a different package (and you need to consider naming) as currently quite confusing!

Reply:
maybe pkg/offline ? with the file writer pieces